x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.ts

```diff
@@ -145,9 +189,67 @@ export const performBulkActionRoute = (
               },
               body: responseBody,
             });
+          case BulkAction.edit:
+            processingResponse = await executeBulkAction(rules.data, async (rule) => {
+              throwHttpError(await mlAuthz.validateRuleType(rule.params.type));
+
+              const editedRule = body[BulkAction.edit].reduce(
+                (acc, action) => appplyBulkActionEditToRule(acc, action),
+                rule
+              );
+
+              const { tags, params: { timelineTitle, timelineId } = {} } = editedRule;
+              const index = 'index' in editedRule.params ? editedRule.params.index : undefined;
+
+              await patchRules({
+                rulesClient,
+                rule,
+                tags,
+                index,
+                timelineTitle,
+                timelineId,
+              });
+            });
+        }
+
+        const errors = processingResponse.results.filter(
+          (resp): resp is RuleActionError => resp?.error !== undefined
+        );
+        const rulesCount = rules.data.length;
+        const errorsCount = errors.length;
+
+        if (errorsCount > 0) {
+          const responseBody = {
+            message: 'Bulk edit partially failed',
+            status_code: 500,
+            errors: errors.map(({ error, rule }) => ({
+              rule_id: rule.id,
+              rule_name: rule.name,
+              error_message: error.message,
+              error_status_code: error.statusCode,
+            })),
+            rules: {
+              total: rulesCount,
+              failed: errorsCount,
+              succeeded: rulesCount - errorsCount,
+            },
+          };
+
+          return response.custom({
+            headers: {
+              'content-type': 'application/json',
+            },
+            body: Buffer.from(JSON.stringify(responseBody)),
+            statusCode: 500,
+          });
```

Comment:
@banderror, I didn't use it because it expects payload as

```typescript
      message: string | Error;
      attributes?: ResponseErrorAttributes;
```

which is differs from what we agreed (doesn't even allow to pass in body status_code, we send with other error responses everywhere in security solution). So I decided to stick with `response.custom`

Reply:
Discussed over zoom.
@vitaliidm will open a ticket for tweaking `SiemResponseFactory.error` method so it could be used for returning arbitrary fields in the response. Then we'll be able to refactor this `response.custom` call.